sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -1407,7 +1441,6 @@ object EliminateSorts extends Rule[LogicalPlan] {
     _.containsPattern(SORT))(applyLocally)
 
   private val applyLocally: PartialFunction[LogicalPlan, LogicalPlan] = {
-    case Sort(_, _, child) if child.maxRows.exists(_ <= 1L) => recursiveRemoveSort(child)
```

Comment:
hmm, are you sure the new rule can fully cover this?

Reply:
I think it is. The `EliminateLimits` only run `Once` , and the added rule run `fixedPoint`. It's no harmful since we have transformWithPruning